### PR TITLE
Mapped some persistence data stuff

### PR DESCRIPTION
**Pull request by liach**
I renamed all persisted to persistent, though vanilla fix types enum calls them saved data.

Mapped out the persistent data manager that is bound to each dimension and the persistent data updater for stuctures.

### CODE REVIEW

File: mappings/net/minecraft/nbt/ChunkUpdateState.mapping [hunk 1]
```diff
@@ -0,0 +1,12 @@
+CLASS ccl net/minecraft/nbt/ChunkUpdateState
```
Comment (asiekierka) [inline mappings/net/minecraft/nbt/ChunkUpdateState.mapping]: nbt?

Comment (asiekierka) [reply mappings/net/minecraft/nbt/ChunkUpdateState.mapping]: Should PersistentState even be in nbt?

File: mappings/net/minecraft/world/PersistentStateContainer.mapping [hunk 2]
```diff
@@ -0,0 +1,8 @@
+CLASS bbx net/minecraft/world/PersistentStateContainer
+	METHOD a getNextAvailableIdForPersistentState (Lbsg;Ljava/lang/String;)I
```
Comment (asiekierka) [inline mappings/net/minecraft/world/PersistentStateContainer.mapping]: (note: bbx is an interface, the verbosity might be warranted)

**Review by asiekierka — CHANGES_REQUESTED** [review]

File: mappings/net/minecraft/world/PersistentStateContainer.mapping (hunk 2 above)
Comment (liach) [reply mappings/net/minecraft/world/PersistentStateContainer.mapping]: This interface is implemented by `World` and `IWorld`. Dunno if this degree of clarity is needed in a class with a ton of stuff (maybe something else can also supply available ids?)

File: mappings/net/minecraft/nbt/ChunkUpdateState.mapping (hunk 1 above)
Comment (liach) [reply mappings/net/minecraft/nbt/ChunkUpdateState.mapping]: Agree. The whole thing should go to world package.

**Comment by liach** [comment]
@asiekierka Can you review again?

**Review by asiekierka — APPROVED** [review]

**Review by shadowfacts — APPROVED** [review]

**Pull request merged**